editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.restconfiguration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.ui.IEditorPart;
+import org.fusesource.ide.camel.editor.CamelEditor;
+import org.fusesource.ide.camel.model.service.core.model.RestElement;
+import org.fusesource.ide.camel.model.service.core.model.RestVerbElement;
+import org.fusesource.ide.camel.test.util.editor.AbstractCamelEditorIT;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * @author brianf
+ *
+ * Load a Camel config with some Rest DSL
+ * See if the Rest tab is there
+ * Switch to the Rest tab <-- stopped here, but the rest is next
+ * See if the first rest element is there (getSelection should by default return it)
+ * Select a second rest element
+ * Switch to Source tab
+ * Switch back to Rest tab
+ * See if the second rest element is still selected
+ * repeat with REST operations
+ * 
+ */
+@RunWith(Parameterized.class)
+public class RestEditorTestIT extends AbstractCamelEditorIT {
+
+	@Parameters(name = "Route container of type: {0}")
+	public static String[] data() {
+	    return new String[] { "camelContext" };
+	}
+	
+	public RestEditorTestIT(String routeContainerType) {
+		this.routeContainerType = routeContainerType;
+	}
+	
+	private void flipTabsToRefreshModel(CamelEditor camelEditor) {
+		// switch to the source tab and back again, which refreshes the REST model
+		camelEditor.setActiveEditor(camelEditor.getSourceEditor());
+		readAndDispatch(20);
+		camelEditor.setActiveEditor(camelEditor.getRestEditor());
+		readAndDispatch(20);
+	}
+	
+	@Test
+	public void openCamelFileMoveToRESTPage() throws Exception {
+		IEditorPart openEditorOnFileStore = openFileInEditor("/rest");
+		assertThat(openEditorOnFileStore).isNotNull();
+		assertThat(openEditorOnFileStore).isInstanceOf(CamelEditor.class);
+
+		readAndDispatch(20);
+
+		// ensure that the REST tab is available and select it
+		assertThat(((CamelEditor)openEditorOnFileStore).getRestEditor()).isNotNull();
+		CamelEditor camelEditor = (CamelEditor)openEditorOnFileStore;
+		camelEditor.setActiveEditor(camelEditor.getRestEditor());
+		assertThat(camelEditor.getActivePage() == CamelEditor.REST_CONF_INDEX);
+		RestConfigEditor restEditor = camelEditor.getRestEditor();
+		camelEditor.setActiveEditor(restEditor);
+
+		readAndDispatch(20);
+
+		// grab the selection from the REST tab, which should be a REST Element
+		StructuredSelection ssel = (StructuredSelection) restEditor.getSelection();
+		assertThat(ssel.isEmpty()).isNotEqualTo(true);
+		assertThat(ssel.getFirstElement() instanceof RestElement).isTrue();
```

Comment:
I think it is easier to read written like that:
assertThat(ssel.getFirstElement()).isInstanceOf(RestElement.class);

but maybe a matter of taste